plugins/filter/ocp_compatibility.py

```diff
@@ -12,22 +17,28 @@ def ocp_compatibility(self, after_api, curr_version, junit_ocp_file):
         from semver import VersionInfo
 
         # convert k8s version to ocp version
-        k8s2ocp = lambda x : f"{int(x.split('.')[0]) + 3}.{int(x.split('.')[1]) - 13}"
+        def k8s2ocp(x):
+            return f"{int(x.split('.')[0]) + 3}.{int(x.split('.')[1]) - 13}"
         # parse the API after the workload installation and write down incompatible OCP versions
-        failed_versions = { k8s2ocp(api['removedInRelease']) for api in after_api }
+        failed_versions = {k8s2ocp(api['removedInRelease'])
+                           for api in after_api}
         version_to_removed_apis = {
-            version: {api['name'] for api in after_api if k8s2ocp(api['removedInRelease']) == version}
+            version: {api['name'] for api in after_api if k8s2ocp(
+                api['removedInRelease']) == version}
             for version in failed_versions
         }
 
         # convert major.minor version to semvers' major.minor.patch
-        ocp2semver = lambda x : VersionInfo.parse(x + ".0")
+        def ocp2semver(x):
+            return VersionInfo.parse(x + ".0")
         # convert semver's major.minor.patch to OCP major.minor
-        semver2ocp = lambda x : '.'.join(str(x).split('.')[:-1])
```

Comment:
Hi, @sshnaidm is lambda-function always a wrong choice in ansible filters?

Reply:
**Assigning** lambda function is wrong according to [PEP-8 ](https://peps.python.org/pep-0008/#programming-recommendations), it triggers [lambda-assignment (E731)](https://docs.astral.sh/ruff/rules/lambda-assignment/) error. Using it _inside_ is not a problem of course.